serialization/counter/marshal_utils.go

```diff
@@ -172,8 +178,14 @@ func EncReflect(v reflect.Value) ([]byte, error) {
 	switch v.Kind() {
 	case reflect.Int, reflect.Int64, reflect.Int32, reflect.Int16, reflect.Int8:
 		return EncInt64(v.Int())
-	case reflect.Uint, reflect.Uint64, reflect.Uint32, reflect.Uint16, reflect.Uint8:
-		return EncUint64(v.Uint())
+	case reflect.Uint32, reflect.Uint16, reflect.Uint8:
+		return encUint64(v.Uint()), nil
+	case reflect.Uint, reflect.Uint64:
+		val := v.Uint()
+		if val > math.MaxInt64 {
+			return nil, fmt.Errorf("failed to marshal counter: value (%T)(%[1]v) out of range", v.Interface())
+		}
+		return encUint64(val), nil
```

Comment:
why not call `EncUint64`?

Reply:
Answer here https://github.com/scylladb/gocql/pull/306#discussion_r1798350361